### PR TITLE
Event api participants

### PR DESCRIPTION
**Pull request by Scandie**
Realization of API for user assignment to event.

### CODE REVIEW

**Comment by sbufan** [comment]
+ Code looks good

**Review by sbufan — COMMENTED** [review]
Code looks good

File: events/event/views.py [hunk 1]
```diff
@@ -1,3 +1,56 @@
...
+    pass
+
+
+class EventUserAssignmentView(View):
+    """
+    View used for user assignment to event.
+    """
+
+    def put(self, request, event_id):
+        """
```
Comment (VKotsovskyi) [inline events/event/views.py]: code says exactly what is written in a comment. No need comment in this case.

File: events/event/views.py [hunk 2]
```diff
@@ -1,3 +1,56 @@
...
+        :param request: request to View
+        :param event_id: id of event for which users will be assigned
+        :return: json containing error message and status 404 if event does not exists.
+        json containing two lists: successfully added users; users which do not exist and status 404
+        if some of assigned users are not existing.
+        json containing list of successfully added users and status 200.
+        """
+        data = json.loads(request.body.decode())
+        not_existing_users, successfully_added = [], []
```
Comment (VKotsovskyi) [inline events/event/views.py]: this works slower than one by one assignment.

File: events/event/views.py [hunk 3]
```diff
@@ -1,3 +1,56 @@
...
+    def put(self, request, event_id):
+        """
+        :param request: request to View
+        :param event_id: id of event for which users will be assigned
+        :return: json containing error message and status 404 if event does not exists.
+        json containing two lists: successfully added users; users which do not exist and status 404
+        if some of assigned users are not existing.
+        json containing list of successfully added users and status 200.
+        """
+        data = json.loads(request.body.decode())
```
Comment (VKotsovskyi) [inline events/event/views.py]: to general name. Add specific names.

File: events/event/views.py [hunk 4]
```diff
@@ -1,3 +1,56 @@
...
+        not_existing_users, successfully_added = [], []
+
+        event = Event.get_by_id(event_id)
+        if not event:
+            return JsonResponse({"error_message": "Such event does not exists"}, status=404)
+
+        for username in data.get('participants'):
```
Comment (VKotsovskyi) [inline events/event/views.py]: This will crash if no key `participants` exists in the request.

File: events/event/views.py [hunk 5]
```diff
@@ -1,3 +1,56 @@
...
+            try:
+                user = User.objects.get(username=username)
+            except User.DoesNotExist:
+                not_existing_users.append(username)
+                continue
+
+            instance, added = EventUserAssignment.objects.get_or_create(user=user, event=event)
+
+            if added:
+                instance.save()
```
Comment (VKotsovskyi) [inline events/event/views.py]: Check if you really need this.

File: events/event/views.py [hunk 6]
```diff
@@ -1,3 +1,56 @@
...
+                successfully_added.append(user.username)
+
+        if not_existing_users:
+
+            return JsonResponse({
+                "not_existing_users": not_existing_users,
+                "successfully_added": successfully_added,
+            }, status=404)
```
Comment (VKotsovskyi) [inline events/event/views.py]: 404 not appropriate status for this. You still add at least some users.

File: events/events/urls.py [hunk 7]
```diff
@@ -21,5 +21,6 @@
 urlpatterns = [
     url(r'^admin/', admin.site.urls),
     url(r'^$', login_required(TemplateView.as_view(template_name='main.html'), login_url='auth/login?next=/')),
-    url(r'^auth/', include('auth.urls', namespace='auth'))
+    url(r'^auth/', include('auth.urls', namespace='auth')),
+    url(r'^api/v1/events/', include('event.urls', namespace='event'))
```
Comment (VKotsovskyi) [inline events/events/urls.py]: I would recommend creating separate file `api_urls` where incluse all urls files from apps, and here include that file with url. Example -  

``` python
url(r'^api/v1/', include('events.apy_urls', namespace='event'))
```

**Review by VKotsovskyi — CHANGES_REQUESTED** [review]
In general looks good, but need some improvements.

File: events/events/urls.py (hunk 7 above)
Comment (Scandie) [reply events/events/urls.py]: @VKotsovskyi I think we should discuss it later and, if needed, add this changes. By now, everyone follows this structure.

File: events/event/models.py [hunk 8]
```diff
@@ -1,17 +1,34 @@
...
+        through_fields=('event', 'user'),
+    )
 
     def __str__(self):
         return "%s" % self.title
 
-    #def save(self, *args, **kwargs):	
-        #super(Event, self).save(*args, **kwargs)
+    @staticmethod
+    def get_all():
```
Comment (yevgenbehen) [inline events/event/models.py]: Please do not use `objects.all()` and do not create `get_all()` method

File: events/event/views.py [hunk 9]
```diff
@@ -1,3 +1,43 @@
...
+
+    def put(self, request, event_id):
+        event_participants = json.loads(request.body.decode())
+        not_existing_users = []
+        successfully_added = []
+        event = Event.get_by_id(event_id)
+        if not event:
+            return JsonResponse({"error_message": "Such event does not exists"}, status=404)
+        if not event_participants.get('participants'):
+            return JsonResponse({"error_message": "Not given any participants"}, status=400)
```
Comment (yevgenbehen) [inline events/event/views.py]: Error message in such cases should sound like "Invalid payload". No details should be provided

File: events/events/settings.py [hunk 10]
```diff
@@ -45,7 +45,7 @@
     'django.middleware.security.SecurityMiddleware',
     'django.contrib.sessions.middleware.SessionMiddleware',
     'django.middleware.common.CommonMiddleware',
-    'django.middleware.csrf.CsrfViewMiddleware',
+    # 'django.middleware.csrf.CsrfViewMiddleware',
```
Comment (yevgenbehen) [inline events/events/settings.py]: Do not commit commented code

**Review by yevgenbehen — CHANGES_REQUESTED** [review]

**Review by VKotsovskyi — APPROVED** [review]

**Review by lhalam — APPROVED** [review]

**Pull request merged**